modules/core/src/main/java/org/locationtech/jts/geom/LinearRing.java

```diff
@@ -132,13 +132,12 @@ protected LinearRing copyInternal() {
     return new LinearRing(points.copy(), factory);
   }
 
-  /** @deprecated */
-  public Geometry reverse()
+  public LinearRing reverse()
   {
-    return super.reverse();
+    return (LinearRing) super.reverse();
   }
 
-  public Geometry reverseInternal() {
+  public LinearRing reverseInternal() {
```

Comment:
I believe this should be protected; otherwise this will be the only reverseInternal which has public scope!

Reply:
Thank you, good catch

Reply:
sorry I missed applying this feedback :(

Reply:
Actually, I think you switched reverse to protected instead of this method.